nautobot/extras/migrations/0024_job_data_migration.py

```diff
@@ -1,17 +1,33 @@
 from django.db import migrations
 
 from nautobot.core.fields import slugify_dots_to_dashes
+from nautobot.extras.constants import (
+    JOB_MAX_GROUPING_LENGTH,
+    JOB_MAX_NAME_LENGTH,
+    JOB_MAX_SLUG_LENGTH,
+)
 
 
 def _create_job_model(job_model_class, class_path):
     try:
         source, module_name, job_class_name = class_path.split("/")
+        # At this point in the migrations, source = "git.RepositoryName" not just "git", so not JOB_MAX_SOURCE_LENGTH
+        if len(source) > 110:
+            print(f'Skipping Job model creation from "{class_path}" as the source is too long')
+            return None
+        if len(module_name) > JOB_MAX_NAME_LENGTH or len(module_name) > JOB_MAX_GROUPING_LENGTH:
```

Comment:
AFAICT We will never hit `len(module_name) > JOB_MAX_GROUPING_LENGTH`.

`JOB_MAX_NAME_LENGTH` = 100. If `module_name` was ever greater than `JOB_MAX_GROUPING_LENGTH` (of 255) then it would fail on the first check.

Reply:
Agreed - I just put this in here as a belt-and-suspenders check in case either constant gets redefined.